source/includes/markdown/_09-rich-text.html.md

```diff
@@ -125,6 +125,76 @@ Here's an example of what a complete rich comment might look like in the API:
 `
 
+## Reading defensively
+
+> Custom handling external media `<object>`
+
+```javascript
+!
+const richText = '<body><object style="display:block" type="application/vnd.asana.external_media" data="https://www.youtube.com/embed/VqnMA3K6-e0"><a href="https://www.youtube.com/embed/VqnMA3K6-e0">https://www.youtube.com/embed/VqnMA3K6-e0</a></object></body>'
+const parser = new DOMParser();
+const richTextDocument = parser.parseFromString(richText, "text/html");
+
+const objects = richTextDocument.querySelectorAll("object");
+
+for (let i = 0; i < objects.length; i++) {
+    replacement = null;
+
+    switch(objects[i].type) {
+        case "application/vnd.asana.external_media":
+            replacement = richTextDocument.createElement('iframe');
+            replacement.width = 420;
+            replacement.height = 315;
+            replacement.src = objects[i].data;
+            break;
+        default:
+            replacement = richTextDocument.createElement('div');
+            replacement.innerHtml = objects[i].innerHTML;
+    }
+
+    if (replacement) {
+        objects[i].parentElement.replaceChild(replacement, objects[i]);
+    }
+}
+
+richTextDocument.body.childNodes.forEach (child => {
+    document.body.append(child);
+});
+```
+
+**We are actively adding new rich text formats to various objects in Asana**. An existing app will break if not built 
+defensively. Apps should use parsers and display logic that is forward compatible with unknown future rich text formats. 
+
+To do this, Asana provides two mechanisms to parse and display tags that the app doesn't explicitly support:
+
+ * Defaults that render in a WebView
+ * Guidelines for how to handle new tags
+
+You can read more about rich text changes in 
+[this forum post](https://forum.asana.com/t/new-rich-text-tags-in-html-notes-and-html-text/115442).
+
+### Render rich text in a WebView
+
+You can expect the rich text HTML to render reasonably in a WebView if you apply the following CSS style to the 
+wrapping DOM node: `overflow-wrap: break-word; white-space: pre-wrap;`. This won't look exactly like it does in Asana, 
+but it will ensure users read it in the same way.
+
+### How to handle new tags (no WebView)
+
+#### An `<object>` with an unhandled type
+Render the `<object>` tag as a block and render the contained HTML with the same behavior as if it were not inside an 
+`<object>`. We will never send an `<object>` tag nested inside another `<object>` tag.
+
+#### An `<img>`
+[Fall back](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attributes) to the alt text if the image can’t 
+be displayed. Render `\n<alt text>\n` since `<img>` are blocks
```

Comment:
We've since updated this to fallback to either the alt text, or the src link, or a combination of both, whichever one is more suitable for your use case.